R/get_tiff_values.R

```diff
@@ -0,0 +1,13 @@
+#' Takes a Spatrast and gives back a matrix with the values of the Spatrast
+#' @param tiff Takes a Spatrast
+
+
+
+get_tiff_values <- function(tiff) {
+    
+    col <- ncol(tiff)
+    row <- nrow(tiff)
+    data <- matrix(values(tiff),col,row,byrow=TRUE)
+    invisible(return(data))
```

Comment:
Finally, instead of returning just the data, I'd like to return the other information that we might need to pass to libtopotoolbox, namely the cell size and the dimensions. We can do that by returning a list:

```R
list(z = terra::values(r,mat=FALSE),
     cellsize = terra::xres(r),
     dims = c(terra::ncol(r), terra::nrow(r)))
```

Note that we use `terra::xres` because we will have validated that the x and y resolutions are the same. And we return the dimensions as `(cols, rows)`, which is how libtopotoolbox expects them, since the data vector is stored in row-major order. This is opposite the way `dim(r)` returns the dimensions.

Reply:
done, but i actually don't understand what xres, yres and nlyr mean and what they are for.

Reply:
`xres` and `yres` are the horizontal and vertical resolutions of the raster grid. They measure how large an area of Earth each pixel in the grid represents. If your raster is in a projected coordinate system, they have units like meters. If it is geographic, they might be in units of degrees of longitude and latitude.

`nlyr` is the number of layers in the raster that you have loaded. Digital elevation models are usually provided as single layer rasters, but you can have more layers. Satellite imagery, for example, often has multiple layers which correspond to the wavelengths of light or colors that the sensor detects.

`terra` has some documentation of these functions that you can access with `help(xres)` or the equivalent, but I can see how that documentation is not incredibly helpful.

Reply:
i knew about the help(xres), but it still left me clueless and i found it to be unhelpful. 
Your explanation  helped me more, thank you!